sunpy/net/dataretriever/tests/test_vsm.py

```diff
@@ -0,0 +1,49 @@
+"""
+This module tests SOLIS VSM Client.
+"""
+#This module was developed with funding provided by
+#the Google Summer of Code 2016.
+import pytest
+
+from astropy import units as u
+from sunpy.net.vso.attrs import Time, Instrument, Physobs, Wavelength
+from sunpy.net.dataretriever.downloader_factory import UnifiedResponse
+from sunpy.net import Fido
+from sunpy.net import attrs as a
+import sunpy.net.dataretriever.sources.vsm as vsm
+
+VClient = vsm.VSMClient()
+
+TRANGE = Time('2014/6/1', '2014/6/4')
+#Not downloading fits files since
+#they are very huge ranging from 75MB-100MB
+@pytest.mark.online
+@pytest.mark.parametrize("time, instrument, wavelength, physobs",
+[(TRANGE, Instrument('vsm'), Wavelength(6302*u.AA),
+Physobs("LOS_MAGNETIC_FIELD"))])
+def test_query(time, instrument, physobs, wavelength):
+    qr = VClient.query(time, instrument, physobs, wavelength)
+    assert len(qr) == 3
+
+def test_can_handle_query():
+    assert VClient._can_handle_query(TRANGE, Instrument('vsm'), Wavelength(6302*u.AA))
+    assert not VClient._can_handle_query(TRANGE, Instrument('vsm'))
+    assert VClient._can_handle_query(TRANGE, Instrument('vsm'), Wavelength(8542*u.AA))
+    assert VClient._can_handle_query(TRANGE, Instrument('vsm'), Wavelength(1083.0*u.nm),
+                                     Physobs('EQUIVALENT_WIDTH'))
```

Comment:
parametrize this test, so `test_can_handle_query` is like:

```python
@pytest.mark.parametrize(" wavelength, physobs, expected", 
                                           [ (wavelength(x), None, True), 
                                             (wavelength(y), None, True), 
                                             (None, None, False), 
                                             (wavelenght(z), phyobs('equivalent_width'), True)]
                        )
def test_can_handle_query(wavelength,  physobs, expected):
    assert VClient._can_handle_query(TRANGE, Instrument('vsm'), wavelength, physobs) is expected
```
 
Reasons for doing so told in a different PR